etc/kayobe/pulp.yml

```diff
@@ -236,806 +227,305 @@ stackhpc_pulp_repo_rocky_9_highavailability_version: "{{ lookup('vars', 'stackhp
-
-  # Additional CentOS Stream 9 repositories
-  - name: "centos-stream-9-nfv-openvswitch-production"
-    base_path: "centos/9-stream/nfv/x86_64/openvswitch-2/production"
-    repository: CentOS Stream 9 - NFV OpenvSwitch
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "centos-stream-9-openstack-yoga-production"
-    base_path: "centos/9-stream/cloud/x86_64/openstack-yoga/production"
-    repository: CentOS Stream 9 - OpenStack Yoga
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "centos-9-opstools-collectd-production"
-    base_path: "centos/9-stream/opstools/x86_64/collectd-5/production"
-    repository: CentOS 9 - OpsTools - collectd
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "centos-stream-9-ceph-pacific-production"
-    base_path: "centos/9-stream/storage/x86_64/ceph-pacific/production"
-    repository: CentOS Stream 9 - Ceph Pacific
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "extra-packages-for-enterprise-linux-9-x86_64-production"
-    repository: Extra Packages for Enterprise Linux 9 - x86_64
-    base_path: "epel/9/Everything/x86_64/production"
-    state: present
-    required: "{{ stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "docker-ce-for-centos-9-production"
-    repository: Docker CE for CentOS 9
-    base_path: "docker-ce/centos/9/x86_64/stable/production"
-    state: present
-    required: "{{ stackhpc_pulp_sync_el_9 | bool }}"
-
-  # Additional RHEL 9 repositories
-  - name: "rhel-9-treasuredata-4-production"
-    base_path: "treasuredata/4/redhat/9/x86_64/production"
-    repository: TreasureData 4 for RHEL 9
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "rhel-9-mariadb-10-6-production"
-    base_path: "mariadb-10.6/yum/rhel/9/x86_64/production"
-    repository: MariaDB 10.6 for RHEL 9
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
-  - name: "rhel-9-influxdb-production"
-    base_path: "influxdb/rhel/9/x86_64/stable/production"
-    repository: InfluxDB for RHEL 9
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+stackhpc_pulp_distribution_rpm_production: >-
+  {%- set prod_dists = [] -%}
+  {%- for repo in stackhpc_pulp_rpm_repos -%}
+  {%- set name = repo.distribution_name ~ 'production' -%}
+  {%- set base_path = repo.base_path ~ 'production' -%}
+  {%- set dev_dist = {
+                      "name": name,
+                      "repository": repo.name,
```

Comment:
For production this should actually be `"distribution": <development distribution>` since we're promoting dev to prod. Only some had this right before though.

Reply:
Added, same on prod deb distributions.